neo/Core/Transaction.cs

```diff
@@ -374,7 +375,32 @@ public virtual bool Verify(IEnumerable<Transaction> mempool)
             }
             if (Attributes.Count(p => p.Usage == TransactionAttributeUsage.ECDH02 || p.Usage == TransactionAttributeUsage.ECDH03) > 1)
                 return false;
+            if (!VerifyReceivingScripts()) return false;
             return this.VerifyScripts();
         }
+
+        private bool VerifyReceivingScripts()
+        {
+            foreach (UInt160 hash in Outputs.Select(p => p.ScriptHash).Distinct())
+            {
+                ContractState contract = Blockchain.Default.GetContract(hash);
+                if (contract == null) continue;
+                using (StateReader service = new StateReader())
+                {
+                    ApplicationEngine engine = new ApplicationEngine(TriggerType.VerificationR, this, Blockchain.Default, service, Fixed8.Zero);
+                    engine.LoadScript(contract.Script, false);
+                    using (ScriptBuilder sb = new ScriptBuilder())
+                    {
+                        sb.EmitPush(0);
+                        sb.Emit(OpCode.PACK);
+                        sb.EmitPush("receiving");
+                        engine.LoadScript(sb.ToArray(), false);
```

Comment:
I believe this will cause contracts that are expecting 2 parameters ( ie `operation` and `args`?) to experience a stack error.

Ideally the number of parameters should be read from the ContractState.  After pushing 'receiving' for the first one, need to push an empty list/bytearray etc afterwards.

Reply:
It won't cause a stack error. Because it pushed two parameters into the stack.

The first is a string "receiving" which is mapped to the parameter `operation`. This is done by the code:
```c#
sb.EmitPush("receiving");
```

The second is an empty array which is mapped to the parameter `args`. This is done by the code:
```c#
sb.EmitPush(0);
sb.Emit(OpCode.PACK);
```

Reply:
I think the issue is that some contracts may accept 1 parameter, and others could take 3.

Reply:
👍